test/SimpleWebRequest.spec.ts

```diff
@@ -88,11 +103,160 @@ describe('SimpleWebRequest', () => {
+            expect(onSuccessHigh).toHaveBeenCalled();
+
+            // and the low priority one gets sent last 
+            jasmine.Ajax.requests.mostRecent().respondWith({ status: 200 });
+            expect(onSuccessLow).toHaveBeenCalled();
+        });
+        
+        it('checks the blocked function again, once the request is on top of the queue', () => {
+            const url = faker.internet.url();
+            const method = 'GET';
+            const blockDefer = SyncTasks.Defer<void>();
+            const onSuccessCritical = jasmine.createSpy('onSuccessCritical');
+            const onSuccessHigh = jasmine.createSpy('onSuccessHigh');
+            const onSuccessHigh2 = jasmine.createSpy('onSuccessHigh2');
+            const blockSpy = jasmine.createSpy('blockSpy').and.callFake(() => blockDefer.promise());
+
+            new SimpleWebRequest<string>(url, method, { priority: WebRequestPriority.Critical }, undefined, blockSpy).start().then(onSuccessCritical);
+            jasmine.clock().tick(10);
+            new SimpleWebRequest<string>(url, method, { priority: WebRequestPriority.High }).start().then(onSuccessHigh);
+            jasmine.clock().tick(10);
+
+            SimpleWebRequestOptions.MaxSimultaneousRequests = 1;
+            // add a new request to kick the queue
+            new SimpleWebRequest<string>(url, method, { priority: WebRequestPriority.High }).start().then(onSuccessHigh2);
+            
+            expect(blockSpy).toHaveBeenCalled();
+
+            jasmine.Ajax.requests.mostRecent().respondWith({ status: 200 });
+            expect(onSuccessHigh).toHaveBeenCalled();
+
+            // unblock the request, it will go back to the queue after the currently executed request
+            blockDefer.resolve(void 0);
+
+            jasmine.Ajax.requests.mostRecent().respondWith({ status: 200 });
+            expect(onSuccessHigh2).toHaveBeenCalled();
+
+            // check if the request at the top of the queue got called again
+            expect(blockSpy).toHaveBeenCalledTimes(2);
+
+            jasmine.Ajax.requests.mostRecent().respondWith({ status: 200 });
+            expect(onSuccessCritical).toHaveBeenCalled();
+        });
+
+        it('fails the request, if the blocking promise rejects', done => {
+            SimpleWebRequestOptions.MaxSimultaneousRequests = 1;
+            const url = faker.internet.url();
+            const method = 'GET';
+            const blockDefer = SyncTasks.Defer<void>();
+            new SimpleWebRequest<string>(url, method, { priority: WebRequestPriority.Critical }, undefined, () => blockDefer.promise()).start()
+            .then(() => fail(), (err: WebErrorResponse) => {
+                expect(err.statusCode).toBe(0);
+                expect(err.statusText).toBe('Error in _blockRequestUntil: Terrible error');
+                done();
+            });
+            
+            blockDefer.reject('Terrible error');
```

Comment:
Suggestion: Put `'Terrible error'` in a constant since it is used in two places. The point of the test is for this and that to be the same; the exact string does not matter.